cookbook/security/api_key_authentication.rst

```diff
@@ -67,7 +60,7 @@ value and then a User object is created::
         public function authenticateToken(TokenInterface $token, UserProviderInterface $userProvider, $providerKey)
         {
             $apiKey = $token->getCredentials();
-            $username = $this->userProvider->getUsernameForApiKey($apiKey);
+            $username = $userProvider->getUsernameForApiKey($apiKey);
```

Comment:
I thought about this again and I think we actually do not need to check for the token type.  The authenticator doesn't do anything special with the token and so it should work just fine.